python/ray/_private/worker.py

```diff
@@ -1478,8 +1478,12 @@ def init(
     if gcs_address is None:
         info_str = "Started a local Ray instance."
     if dashboard_url:
+        if not (
```

Comment:
This protocol changing here is not ideal, and if you do this you might also need to change https://github.com/ray-project/ray/blob/master/python/ray/client_builder.py#L88

I currently don't have a proposal to do it better (since we cannot change the semantics of the dashboard_url parameter either). If others have ideas please chime in!

Reply:
can you use `urllib.parse.urlparse` and `urllib.parse.unparse` to do that?

Reply:
can you use `from urllib.parse import urlparse, urlunparse` to achieve the same goal?

Reply:
What about

1. When reporting dashboard URL from the dashboard, we include protocol there. (use urlparse and append https if there's no protocol).
2. When `address_info["webui_url"]` is constructed, we remove protocol. (so we keep the backward compatibility)
3. Add a new field to `address_info` either (1) `webui_url_protocol` or `webui_url_with_protocol`
?

Reply:
Updated to make `webui_url_with_protocol` a property of the private Node class, so we don't need to change the protocol of `webui_url` and `dashboard_url`